scripts/operations/node_management/encryption.sh

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/env sh
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+_base=$(basename "$0")
+_dir=$(cd -P -- "$(dirname -- "$(command -v -- "$0")")" && pwd -P || exit 126)
+export _base _dir
+set "${SETOPTS:--eu}"
+
+tmpfiles=
+cleanup() {
+  if [ "" != "${tmpfiles}" ]; then
+    rm -fr "${tmpfiles}"
+  fi
+}
+
+trap cleanup EXIT
+
+# Secret data length must be 16,24, or 32 bytes long or its invalid A valid
+# secrets' data can only be 16, 24, or 32 bytes long according to the k8s docs and testing. If we provide data of any length that isn't these lengths k8s will *not* start.
+validatelen() {
+  len=$(echo "$@" | awk '{print length}')
+  if [ 16 -eq "${len}" ] \
+    || [ 24 -eq "${len}" ] \
+    || [ 32 -eq "${len}" ]; then
+    return "${len}"
+  fi
+  return 0
+}
+
+# Allow aescbc or aesgcm provider types only
+validprovider() {
+  provider=${1:-invalid}
+  if [ "aesgcm" = "${provider}" ] || [ "aescbc" = "${provider}" ]; then
+    return 1
+  fi
+  return 0
+}
+
+# Validate a secret type, identity is a special case
+validate() {
```

Comment:
I think I must be missing something here. I don't see this function ever being called in this script, and it seems odd to make a function which just returns 1. It feels like either:
1. This script sources some other shell file, and that file calls this function
2. Some other script sources THIS script, and that file calls this function.

But I don't see this script sourcing anything else. I didn't check to see if the other scripts in this PR sourced this file, but I do see that they never call the validate function that I could see. 

What am I missing?

Reply:
Probably leftovers from a red/green refactor, I'll nuke it.